docs: improved docs around listener config (#5373)
confluentinc/ksql: docs/operate-and-deploy/installation/server-config/config-reference.md

[ksqlDB Server image](https://hub.docker.com/r/confluentinc/ksqldb-server/)
is `KSQL_LISTENERS`.

### ksql.internal.listener

The `ksql.internal.listener` setting controls the address bound for use by internal,
intra-cluster communication.

> If not set, the internal listener default to the first listener defined by `listeners`.

Reviewer: ```suggestion
If not set, the internal listener defaults to the first listener defined by `listeners`.
```

This setting is most often useful in a IaaS environment to separate external-facing
traffic from internal traffic.

### ksql.advertised.listener

> This is the url used for inter-node communication.  Unlike `listeners` or `ksql.internal.listener`,

Reviewer: ```suggestion
This is the URL used for inter-node communication.  Unlike `listeners` or `ksql.internal.listener`,
```

this configuration doesn't create a listener. Instead, it is used to set an externally routable

> url that other ksqlDB nodes will use to communicate with this node. It only needs to be set if

Reviewer: ```suggestion
URL that other ksqlDB nodes will use to communicate with this node. It only needs to be set if
```

the internal listener is not externally resolvable or routable.

> If not set, the default is to use the internal listener, as controlled by `ksql.internal.listener`.

Reviewer: ```suggestion
If not set, the default behavior is to use the internal listener, which is controlled by `ksql.internal.listener`.
```

> If the `ksql.internal.listener` resolves, either by being explicitly set or from the default first

Reviewer: ```suggestion
If the `ksql.internal.listener` resolves to a URL that uses `localhost`, either by being explicitly set or from the default first
```

> url in `listeners`, to a url using `localhost`, a wildcard ip such as `0.0.0.0`, or a hostname that

Reviewer: ```suggestion
URL in `listeners`, to a URL using `localhost`, a wildcard IP address, like `0.0.0.0`, or a host name that
```

> url in `listeners`, to a url using `localhost`, a wildcard ip such as `0.0.0.0`, or a hostname that

Reviewer: ```suggestion
set `ksql.advertised.listeners` to a URL that ksqlDB nodes can resolve.
```

> other ksqlDB nodes can either not resolve, or can not route requests to, then

Reviewer: ```suggestion

```

> `ksql.advertised.listeners` should be set to a url that can.

Reviewer: ```suggestion

```

> See [Configuring Listeners of a ksqlDB Cluster](./index.md#configuringlisteners-of-a-ksqldb-cluster)

Reviewer: ```suggestion
See [Configuring Listeners of a ksqlDB Cluster](./index.md#configuring-listeners-of-a-ksqldb-cluster)
```

> for more info.

Reviewer: ```suggestion

```
